subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1220,6 +1220,100 @@ public FileCollection getFiles() {
                 return new ConfigurationFileCollection(Specs.<Dependency>satisfyAll(), viewAttributes, componentFilter, lenient, allowNoMatchingVariants);
             }
         }
+
+        private class LenientResolutionResult implements ResolutionResult {
```

Comment:
It's a bit complicated to set the error handler at construction time, because the resolution result is created _before_ we ask for it. I'll see what I can do, but I already tried this.

Reply:
Done, with the drawback that it's no longer on a public API, so not usable in IDE import, for example.